File: dashboard/src/types/agent.ts
/**
 * 智能体状态枚举
 */
export enum AgentStatus {
  Active = 'active',
  Inactive = 'inactive',
}

/**
 * 智能体状态显示名称映射
 */
export const AgentStatusLabels: Record<AgentStatus, string> = {
  [AgentStatus.Active]: '活跃',
  [AgentStatus.Inactive]: '停用',
}

/**
 * 智能体状态选项（用于下拉选择）
 */
export const AgentStatusOptions = [
  { value: AgentStatus.Active, label: AgentStatusLabels[AgentStatus.Active] },
  { value: AgentStatus.Inactive, label: AgentStatusLabels[AgentStatus.Inactive] },
]

/**
 * 外部智能体类型枚举
 */
export enum ExternalType {
  Dify = 'dify',
}

/**
 * 外部智能体类型显示名称映射
 */
export const ExternalTypeLabels: Record<ExternalType, string> = {
  [ExternalType.Dify]: 'Dify',
}

/**
 * 外部智能体类型选项
 */
export const ExternalTypeOptions = [
  { value: ExternalType.Dify, label: ExternalTypeLabels[ExternalType.Dify] },
]

/**
 * Dify 配置
 */
export interface DifyConfig {
  base_url: string
  api_key?: string // 创建/更新时传入
  api_key_masked?: string // 响应时返回
}

/**
 * 供应商简要信息（Agent 关联）
 */
export interface ProviderBrief {
  id: string
  name: string
  provider_type: string
  base_url: string
}

/**
 * 技能简要信息
 */
export interface SkillBrief {
  id: string
  name: string
  description: string
}

/**
 * 智能体配置
 */
export interface Agent {
  id: string
  name: string
  description: string
  provider_id: string
  provider?: ProviderBrief
  model: string
  system_prompt: string
  temperature: number
  max_iterations: number
  max_output_tokens: number
  working_dir: string
  compression_ratio: number
  compression_keep_size: number
  context_max_tokens: number
  status: AgentStatus
  // 是否使用所有技能
  use_all_skills: boolean
  // 关联的技能列表
  skills?: SkillBrief[]
  // 是否使用所有工具
  use_all_tools: boolean
  // 关联的工具名称列表
  tool_names?: string[]
  // 是否可作为子智能体被调用
  callable: boolean
  // 作为子智能体时的工具描述
  callable_description: string
  // 是否启用 MCP 工具
  enable_mcp: boolean
  // 是否为外部智能体
  external: boolean
  // 外部智能体类型
  external_type?: ExternalType
  // 外部智能体配置
  external_config?: DifyConfig
  created_at: string
  updated_at: string
}

/**
 * 智能体默认配置
 */
export interface AgentDefaults {
  system_prompt: string
  temperature: number
  max_iterations: number
  max_output_tokens: number
  compression_ratio: number
  compression_keep_size: number
  context_max_tokens: number
}

/**
 * 校验结果
 */
export interface ValidationResult {
  valid: boolean
  errors: string[]
}

/**
 * 创建智能体请求（简化版，只需名称和描述）
 */
export interface CreateAgentRequest {
  name: string
  description?: string
  // 以下字段可选，创建时可留空
  provider_id?: string
  model?: string
  system_prompt?: string
  working_dir?: string
  temperature?: number
  max_iterations?: number
  max_output_tokens?: number
  compression_ratio?: number
  compression_keep_size?: number
  context_max_tokens?: number
  // 技能配置
  use_all_skills?: boolean
  skill_ids?: string[]
  // 工具配置
  use_all_tools?: boolean
  tool_names?: string[]
  // 子智能体配置
  callable?: boolean
  callable_description?: string
  // MCP 配置
  enable_mcp?: boolean
  // 外部智能体配置
  external?: boolean
  external_type?: ExternalType
  external_config?: DifyConfig
}

/**
 * 更新智能体请求
 */
export interface UpdateAgentRequest {
  name?: string
  description?: string
  provider_id?: string
  model?: string
  system_prompt?: string
  temperature?: number
  max_iterations?: number
  max_output_tokens?: number
  working_dir?: string
  compression_ratio?: number
  compression_keep_size?: number
  context_max_tokens?: number
  status?: AgentStatus
  // 技能配置
  use_all_skills?: boolean
  skill_ids?: string[]
  // 工具配置
  use_all_tools?: boolean
  tool_names?: string[]
  // 子智能体配置
  callable?: boolean
  callable_description?: string
  // MCP 配置
  enable_mcp?: boolean
  // 外部智能体配置（仅外部智能体可用）
  external_config?: DifyConfig
}

/**
 * 更新智能体技能配置请求
 */
export interface UpdateAgentSkillsRequest {
  use_all_skills: boolean
  skill_ids?: string[]
}

/**
 * 可用工具列表
 */
export const AvailableTools = [
  { name: 'list_dir', label: '列出目录' },
  { name: 'read_file', label: '读取文件' },
  { name: 'write_file', label: '写入文件' },
  { name: 'edit_file', label: '编辑文件' },
  { name: 'shell', label: '执行命令' },
  { name: 'todo_write', label: '任务管理' },
  { name: 'cron', label: '定时任务' },
] as const

File: beepbot/internal/types/agent.go
package types

import (
	"time"
)

// AgentStatus 智能体状态
type AgentStatus string

const (
	AgentStatusActive   AgentStatus = "active"
	AgentStatusInactive AgentStatus = "inactive"
)

// Agent 智能体配置
type Agent struct {
	ID                  string      `json:"id" gorm:"column:id;primaryKey;type:varchar(64)"`
	Name                string      `json:"name" gorm:"column:name;type:varchar(128);not null;uniqueIndex"`
	Description         string      `json:"description" gorm:"column:description;type:text"`
	ProviderID          string      `json:"provider_id" gorm:"column:provider_id;type:varchar(64);index"` // 可为空，编辑时配置
	Model               string      `json:"model" gorm:"column:model;type:varchar(128)"`                  // 可为空，编辑时配置
	SystemPrompt        string      `json:"system_prompt" gorm:"column:system_prompt;type:text"`
	Temperature         float32     `json:"temperature" gorm:"column:temperature;type:real;default:0.7"`
	MaxIterations       int         `json:"max_iterations" gorm:"column:max_iterations;type:integer;default:50"`
	MaxOutputTokens     int64       `json:"max_output_tokens" gorm:"column:max_output_tokens;type:integer;default:4096"`
	WorkingDir          string      `json:"working_dir" gorm:"column:working_dir;type:varchar(512)"` // 可为空，自动生成或编辑时配置
	CompressionRatio    float64     `json:"compression_ratio" gorm:"column:compression_ratio;type:real;default:0.7"`
	CompressionKeepSize int         `json:"compression_keep_size" gorm:"column:compression_keep_size;type:integer;default:5"`
	ContextMaxTokens    int64       `json:"context_max_tokens" gorm:"column:context_max_tokens;type:integer;default:4096"`
	Status              AgentStatus `json:"status" gorm:"column:status;type:varchar(16);default:active;index"`
	// UseAllSkills 是否使用系统所有技能
	// true: 使用所有技能（默认）
	// false: 仅使用关联表中的技能
	UseAllSkills bool `json:"use_all_skills" gorm:"column:use_all_skills;type:boolean;default:true"`
	// UseAllTools 是否使用所有工具
	// true: 使用所有工具（默认）
	// false: 仅使用关联表中的工具
	UseAllTools bool `json:"use_all_tools" gorm:"column:use_all_tools;type:boolean;default:true"`
	// Callable 是否可作为子智能体被调用
	Callable bool `json:"callable" gorm:"column:callable;type:boolean;default:false;index"`
	// CallableDescription 作为子智能体时的工具描述（供 LLM 理解用途）
	CallableDescription string    `json:"callable_description" gorm:"column:callable_description;type:text"`
	CreatedAt           time.Time `json:"created_at" gorm:"column:created_at;type:timestamptz;not null"`
	UpdatedAt           time.Time `json:"updated_at" gorm:"column:updated_at;type:timestamptz;not null"`
}

// AgentQuery 智能体查询参数
type AgentQuery struct {
	Name   string      // 名称模糊搜索
	Status AgentStatus // 状态筛选
}

// TableName 指定表名
func (Agent) TableName() string {
	return "agents"
}

// UsageStatsPoint 单个时间点的统计数据
type UsageStatsPoint struct {
	Time          time.Time `json:"time"`
	SessionCount  int64     `json:"session_count"`
	MessageCount  int64     `json:"message_count"`
	InputTokens   int64     `json:"input_tokens"`
	OutputTokens  int64     `json:"output_tokens"`
	TotalTokens   int64     `json:"total_tokens"`
}

// UsageStatsResponse 用量统计响应
type UsageStatsResponse struct {
	Points []UsageStatsPoint `json:"points"`
}
